tm2/pkg/bft/privval/signer/remote/client/options.go

```diff
@@ -0,0 +1,143 @@
+package client
+
+import (
+	"fmt"
+	"log/slog"
+	"net"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/gnolang/gno/tm2/pkg/crypto/ed25519"
+	osm "github.com/gnolang/gno/tm2/pkg/os"
+)
+
+// RemoteSignerClient implements types.Signer by connecting to a RemoteSignerServer.
+type RemoteSignerClient struct {
```

Comment:
Why define this in options?

Reply:
It seemed logical to me to group the constructor based on options, the functional methods to modify the options, and the associated struct in the `options.go` file, and to put the struct and the constructor related to the config (in the sense of the node config meant to be marshalled in TOML) in the `config.go` file. 
But I'm totally ok if you want me to move it somewhere else, just let me know where.

Reply:
I automatically went to `client.go` to find the declaration, so most probably there. What do you think?

Reply:
Done https://github.com/gnolang/gno/pull/3554/commits/d41f846e8c718f8a5fb7238fb035d75407393c5b